### ISSUE TITLE
ARC: tests/kernel/critical/kernel.common times out on nsim_em and nsim_sem

### ISSUE THREAD

**Issue by andrewboie** [zephyrproject-rtos/zephyr#14641]
Test case just hangs on nsim_em and nsim_sem simulator:

```
***** Booting Zephyr OS v1.14.0-rc1-1250-g4e097b4d541f *****
Running test suite kernel_critical
===================================================================
starting test - test_critical
```

**Comment by andrewboie** [comment]
It looks like the test eventually completes, but it takes a long time, longer than the allowed timeout in sanitycheck for the test.

On qemu_x86, the test takes about 16 seconds.
On nsim_em, the test takes about 105 seconds on my machine.

**Comment by vonhust** [comment]
This may be fixed by #14558, where i used 10 Mhz  host timer as timer resource. 

If the timer is accurate,  the critical test will run 1500-1600 ms.

**Comment by nashif** [comment]
this was already fixed by #14558
